src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,173 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public class MyNode<T> {
+        private T value;
+        private MyNode<T> prev;
+        private MyNode<T> next;
+
+        MyNode(T x) {
+            value = x;
+        }
```

Comment:
The `MyNode` constructor should initialize all fields (`value`, `prev`, `next`). Currently, it only initializes `value`. Consider adding parameters for `prev` and `next` and initializing them here.